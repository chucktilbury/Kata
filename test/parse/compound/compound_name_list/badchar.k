;-------------------------------------------------------------------------------
; Demonstrate that a bad ',' causes an error
;-------------------------------------------------------------------------------

func name ()() {

    func_ref()(this.is.a.name + whoopd, asd.qwe.rt)
}
