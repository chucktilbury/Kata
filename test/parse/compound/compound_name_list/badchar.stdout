TRACE: main(): push trace state: true
ENTER: main(): 
    ENTER: init_parser(): 
        TRACE: init_parser(): file name: "/home/chuck/Src/Kata/test/parse/compound/compound_name_list/badchar.k"
        ENTER: init_paths(): 
            ENTER: add_path(): 
                TRACE: add_path(): ./
            RETURN: add_path(): 
            ENTER: add_path(): 
                TRACE: add_path(): ../
            RETURN: add_path(): 
            ENTER: add_path(): 
                TRACE: add_path(): ../test/
            RETURN: add_path(): 
            ENTER: add_path(): 
                TRACE: add_path(): ../test/inputs/
            RETURN: add_path(): 
            ENTER: add_path(): 
                TRACE: add_path(): ../test/misc/
            RETURN: add_path(): 
            ENTER: add_path(): 
                TRACE: add_path(): ../test/parse/
            RETURN: add_path(): 
            ENTER: add_path(): 
                TRACE: add_path(): ../test/stuff/
            RETURN: add_path(): 
        RETURN: init_paths(): 
        ENTER: find_file(): 
            TRACE: find_file(): trying: /home/chuck/Src/Kata/test/parse/compound/compound_name_list/badchar.k
            TRACE: find_file(): found: /home/chuck/Src/Kata/test/parse/compound/compound_name_list/badchar.k
        RETURN(create_string(tmp_buf)): find_file() - (!NULL)
        ENTER: init_scanner(): 
            ENTER: open_file(): 
                ENTER: push_input_file(): 
                    TRACE: push_input_file(): input file: /home/chuck/Src/Kata/test/parse/compound/compound_name_list/badchar.k
                RETURN: push_input_file(): 
            RETURN: open_file(): 
        RETURN: init_scanner(): 
        ENTER: create_parser_state(): 
            ENTER: push_scope(): 
                TRACE: push_scope(): scope: PRIVATE
            RETURN: push_scope(): 
            ENTER: push_name(): 
                TRACE: push_name(): name: root
            RETURN: push_name(): 
        RETURN(parser_state): create_parser_state() - (!NULL)
    RETURN: init_parser(): 
    TRACE: main(): --------------------------------------------
    TRACE: main(): Begin Parse
    ENTER: parse(): 
        ENTER: parse_module(): 
            ENTER: parse_module_item(): 
                ENTER: post_token_queue(): 
                RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                ENTER: parse_namespace_item(): 
                    ENTER: post_token_queue(): 
                    RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                    ENTER: parse_scope_operator(): 
                    RETURN(node): parse_scope_operator() - (NULL)
                    ENTER: parse_namespace_definition(): 
                    RETURN(node): parse_namespace_definition() - (NULL)
                    ENTER: parse_class_definition(): 
                    RETURN(node): parse_class_definition() - (NULL)
                    ENTER: parse_function_definition(): 
                        ENTER: post_token_queue(): 
                        RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                        ENTER: advance_token(): 
                        RETURN(tqueue->crnt->tok): advance_token() - (!NULL)
                        ENTER: parse_compound_name(): 
                            ENTER: post_token_queue(): 
                            RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                            TERM: parse_compound_name(): symbol: "name": 5: 10: "badchar.k"
                            ENTER: advance_token(): 
                            RETURN(tqueue->crnt->tok): advance_token() - (!NULL)
                        RETURN(node): parse_compound_name() - (!NULL)
                        ENTER: parse_var_decl_list(): 
                            ENTER: post_token_queue(): 
                            RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                            ENTER: advance_token(): 
                            RETURN(tqueue->crnt->tok): advance_token() - (!NULL)
                            ENTER: parse_var_decl(): 
                                ENTER: parse_type_name(): 
                                    ENTER: parse_literal_type_name(): 
                                    RETURN(node): parse_literal_type_name() - (NULL)
                                    ENTER: parse_compound_name(): 
                                        ENTER: post_token_queue(): 
                                        RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                        ENTER: reset_token_queue(): 
                                        RETURN: reset_token_queue(): 
                                    RETURN(node): parse_compound_name() - (NULL)
                                RETURN(node): parse_type_name() - (NULL)
                            RETURN(node): parse_var_decl() - (NULL)
                            ENTER: advance_token(): 
                            RETURN(tqueue->crnt->tok): advance_token() - (!NULL)
                        RETURN(node): parse_var_decl_list() - (!NULL)
                        ENTER: parse_var_decl_list(): 
                            ENTER: post_token_queue(): 
                            RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                            ENTER: advance_token(): 
                            RETURN(tqueue->crnt->tok): advance_token() - (!NULL)
                            ENTER: parse_var_decl(): 
                                ENTER: parse_type_name(): 
                                    ENTER: parse_literal_type_name(): 
                                    RETURN(node): parse_literal_type_name() - (NULL)
                                    ENTER: parse_compound_name(): 
                                        ENTER: post_token_queue(): 
                                        RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                        ENTER: reset_token_queue(): 
                                        RETURN: reset_token_queue(): 
                                    RETURN(node): parse_compound_name() - (NULL)
                                RETURN(node): parse_type_name() - (NULL)
                            RETURN(node): parse_var_decl() - (NULL)
                            ENTER: advance_token(): 
                            RETURN(tqueue->crnt->tok): advance_token() - (!NULL)
                        RETURN(node): parse_var_decl_list() - (!NULL)
                        ENTER: parse_function_body(): 
                            ENTER: post_token_queue(): 
                            RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                            ENTER: advance_token(): 
                            RETURN(tqueue->crnt->tok): advance_token() - (!NULL)
                            ENTER: parse_function_body_element(): 
                                ENTER: post_token_queue(): 
                                RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                ENTER: parse_var_definition(): 
                                    ENTER: post_token_queue(): 
                                    RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                    ENTER: reset_token_queue(): 
                                    RETURN: reset_token_queue(): 
                                RETURN(node): parse_var_definition() - (NULL)
                                ENTER: parse_function_reference(): 
                                    ENTER: post_token_queue(): 
                                    RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                    ENTER: parse_compound_reference(): 
                                        ENTER: post_token_queue(): 
                                        RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                        ENTER: parse_compound_ref_item(): 
                                            ENTER: post_token_queue(): 
                                            RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                            TERM: parse_compound_ref_item(): symbol: "func_ref": 7: 13: "badchar.k"
                                            ENTER: advance_token(): 
                                            RETURN(tqueue->crnt->tok): advance_token() - (!NULL)
                                        RETURN(node): parse_compound_ref_item() - (!NULL)
                                        ENTER: finalize_token_queue(): 
                                        RETURN: finalize_token_queue(): 
                                    RETURN(node): parse_compound_reference() - (!NULL)
                                    ENTER: parse_expression_list(): 
                                        ENTER: post_token_queue(): 
                                        RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                        TRACE: parse_expression_list(): state: 0, len: 0
                                        ENTER: advance_token(): 
                                        RETURN(tqueue->crnt->tok): advance_token() - (!NULL)
                                        TRACE: parse_expression_list(): state: 1, len: 0
                                        ENTER: parse_expression(): 
                                            ENTER: post_token_queue(): 
                                            RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                            TRACE: parse_expression(): state: 0, stack: 0, queue: 0
                                            ENTER: parse_operator(): 
                                            RETURN(node): parse_operator() - (!NULL)
                                            TRACE: parse_expression(): state: 101, stack: 0, queue: 0
                                            ENTER: reset_token_queue(): 
                                            RETURN: reset_token_queue(): 
                                        RETURN(node): parse_expression() - (NULL)
                                        ENTER: advance_token(): 
                                        RETURN(tqueue->crnt->tok): advance_token() - (!NULL)
                                        TRACE: parse_expression_list(): state: 100, len: 0, pcount: 0
                                        ENTER: finalize_token_queue(): 
                                        RETURN: finalize_token_queue(): 
                                    RETURN(node): parse_expression_list() - (!NULL)
                                    ENTER: parse_compound_name_list(): 
                                        TRACE: parse_compound_name_list(): state = 0
                                        TERM: parse_compound_name_list(): (: "(": 7: 16: "badchar.k"
                                        ENTER: advance_token(): 
                                        RETURN(tqueue->crnt->tok): advance_token() - (!NULL)
                                        TRACE: parse_compound_name_list(): state = 1
                                        TERM: parse_compound_name_list(): symbol: "this": 7: 20: "badchar.k"
                                        ENTER: parse_compound_name(): 
                                            ENTER: post_token_queue(): 
                                            RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                            TERM: parse_compound_name(): symbol: "this": 7: 20: "badchar.k"
                                            ENTER: advance_token(): 
                                            RETURN(tqueue->crnt->tok): advance_token() - (!NULL)
                                            ENTER: advance_token(): 
                                            RETURN(tqueue->crnt->tok): advance_token() - (!NULL)
                                            TERM: parse_compound_name(): symbol: "is": 7: 23: "badchar.k"
                                            ENTER: advance_token(): 
                                            RETURN(tqueue->crnt->tok): advance_token() - (!NULL)
                                            ENTER: advance_token(): 
                                            RETURN(tqueue->crnt->tok): advance_token() - (!NULL)
                                            TERM: parse_compound_name(): symbol: "a": 7: 25: "badchar.k"
                                            ENTER: advance_token(): 
                                            RETURN(tqueue->crnt->tok): advance_token() - (!NULL)
                                            ENTER: advance_token(): 
                                            RETURN(tqueue->crnt->tok): advance_token() - (!NULL)
                                            TERM: parse_compound_name(): symbol: "name": 7: 30: "badchar.k"
                                            ENTER: advance_token(): 
                                            RETURN(tqueue->crnt->tok): advance_token() - (!NULL)
                                        RETURN(node): parse_compound_name() - (!NULL)
                                        TRACE: parse_compound_name_list(): state = 2
                                        TERM: parse_compound_name_list(): +: "+": 7: 32: "badchar.k"
(: "(": 7: 16: "badchar.k"
symbol: "this": 7: 20: "badchar.k"
.: ".": 7: 21: "badchar.k"
symbol: "is": 7: 23: "badchar.k"
.: ".": 7: 24: "badchar.k"
symbol: "a": 7: 25: "badchar.k"
.: ".": 7: 26: "badchar.k"
symbol: "name": 7: 30: "badchar.k"
+: "+": 7: 32: "badchar.k"
                                        TRACE: parse_compound_name_list(): state = 102
                                    RETURN(node): parse_compound_name_list() - (NULL)
(: "(": 7: 16: "badchar.k"
symbol: "this": 7: 20: "badchar.k"
.: ".": 7: 21: "badchar.k"
symbol: "is": 7: 23: "badchar.k"
.: ".": 7: 24: "badchar.k"
symbol: "a": 7: 25: "badchar.k"
.: ".": 7: 26: "badchar.k"
symbol: "name": 7: 30: "badchar.k"
+: "+": 7: 32: "badchar.k"
                                RETURN(node): parse_function_reference() - (NULL)
                                ENTER: parse_create_reference(): 
                                    ENTER: post_token_queue(): 
                                    RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                    ENTER: parse_create_name(): 
                                        ENTER: post_token_queue(): 
                                        RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                        ENTER: reset_token_queue(): 
                                        RETURN: reset_token_queue(): 
                                    RETURN(node): parse_create_name() - (NULL)
                                    ENTER: reset_token_queue(): 
                                    RETURN: reset_token_queue(): 
                                RETURN(node): parse_create_reference() - (NULL)
                                ENTER: parse_destroy_reference(): 
                                    ENTER: post_token_queue(): 
                                    RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                    ENTER: parse_destroy_name(): 
                                        ENTER: post_token_queue(): 
                                        RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                        ENTER: reset_token_queue(): 
                                        RETURN: reset_token_queue(): 
                                    RETURN(node): parse_destroy_name() - (NULL)
                                    ENTER: reset_token_queue(): 
                                    RETURN: reset_token_queue(): 
                                RETURN(node): parse_destroy_reference() - (NULL)
                                ENTER: parse_assignment(): 
                                    ENTER: post_token_queue(): 
                                    RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                    TRACE: parse_assignment(): state = 0
                                    ENTER: parse_compound_reference(): 
                                        ENTER: post_token_queue(): 
                                        RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                        ENTER: parse_compound_ref_item(): 
                                            ENTER: post_token_queue(): 
                                            RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                            ENTER: parse_array_reference(): 
                                                ENTER: post_token_queue(): 
                                                RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                                ENTER: reset_token_queue(): 
                                                RETURN: reset_token_queue(): 
                                            RETURN(node): parse_array_reference() - (NULL)
                                            ENTER: reset_token_queue(): 
                                            RETURN: reset_token_queue(): 
                                        RETURN(node): parse_compound_ref_item() - (NULL)
                                        ENTER: reset_token_queue(): 
                                        RETURN: reset_token_queue(): 
                                    RETURN(node): parse_compound_reference() - (NULL)
                                    TRACE: parse_assignment(): state = 101
                                    ENTER: reset_token_queue(): 
                                    RETURN: reset_token_queue(): 
                                RETURN(node): parse_assignment() - (NULL)
                                ENTER: parse_while_clause(): 
                                    ENTER: post_token_queue(): 
                                    RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                    ENTER: parse_while_definition(): 
                                        ENTER: post_token_queue(): 
                                        RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                        ENTER: reset_token_queue(): 
                                        RETURN: reset_token_queue(): 
                                    RETURN(node): parse_while_definition() - (NULL)
                                    ENTER: reset_token_queue(): 
                                    RETURN: reset_token_queue(): 
                                RETURN(node): parse_while_clause() - (NULL)
                                ENTER: parse_do_clause(): 
                                    ENTER: post_token_queue(): 
                                    RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                    ENTER: reset_token_queue(): 
                                    RETURN: reset_token_queue(): 
                                RETURN(node): parse_do_clause() - (NULL)
                                ENTER: parse_for_clause(): 
                                    ENTER: post_token_queue(): 
                                    RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                    TRACE: parse_for_clause(): state = 0
                                    TRACE: parse_for_clause(): state = 101
                                    ENTER: reset_token_queue(): 
                                    RETURN: reset_token_queue(): 
                                RETURN(node): parse_for_clause() - (NULL)
                                ENTER: parse_if_clause(): 
                                RETURN(node): parse_if_clause() - (NULL)
                                ENTER: parse_try_clause(): 
                                RETURN(node): parse_try_clause() - (NULL)
                                ENTER: parse_switch_clause(): 
                                RETURN(node): parse_switch_clause() - (NULL)
                                ENTER: parse_break_statement(): 
                                RETURN(node): parse_break_statement() - (NULL)
                                ENTER: parse_continue_statement(): 
                                RETURN(node): parse_continue_statement() - (NULL)
                                ENTER: parse_trace_statement(): 
                                RETURN(node): parse_trace_statement() - (NULL)
                                ENTER: parse_inline_statement(): 
                                RETURN(node): parse_inline_statement() - (NULL)
                                ENTER: parse_yield_statement(): 
                                RETURN(node): parse_yield_statement() - (NULL)
                                ENTER: parse_type_statement(): 
                                RETURN(node): parse_type_statement() - (NULL)
                                ENTER: parse_exit_statement(): 
                                RETURN(node): parse_exit_statement() - (NULL)
                                ENTER: parse_print_statement(): 
                                RETURN(node): parse_print_statement() - (NULL)
                                ENTER: parse_return_statement(): 
                                RETURN(node): parse_return_statement() - (NULL)
                                ENTER: parse_raise_statement(): 
                                RETURN(node): parse_raise_statement() - (NULL)
                                ENTER: parse_function_body(): 
                                    ENTER: post_token_queue(): 
                                    RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                                    ENTER: reset_token_queue(): 
                                    RETURN: reset_token_queue(): 
                                RETURN(node): parse_function_body() - (NULL)
                                ENTER: reset_token_queue(): 
                                RETURN: reset_token_queue(): 
                            RETURN(node): parse_function_body_element() - (NULL)
(: "(": 7: 16: "badchar.k"
symbol: "this": 7: 20: "badchar.k"
.: ".": 7: 21: "badchar.k"
symbol: "is": 7: 23: "badchar.k"
.: ".": 7: 24: "badchar.k"
symbol: "a": 7: 25: "badchar.k"
.: ".": 7: 26: "badchar.k"
symbol: "name": 7: 30: "badchar.k"
+: "+": 7: 32: "badchar.k"
                        RETURN(node): parse_function_body() - (NULL)
(: "(": 7: 16: "badchar.k"
symbol: "this": 7: 20: "badchar.k"
.: ".": 7: 21: "badchar.k"
symbol: "is": 7: 23: "badchar.k"
.: ".": 7: 24: "badchar.k"
symbol: "a": 7: 25: "badchar.k"
.: ".": 7: 26: "badchar.k"
symbol: "name": 7: 30: "badchar.k"
+: "+": 7: 32: "badchar.k"
                    RETURN(node): parse_function_definition() - (NULL)
                    ENTER: parse_create_definition(): 
                        ENTER: post_token_queue(): 
                        RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                        ENTER: reset_token_queue(): 
                        RETURN: reset_token_queue(): 
                    RETURN(node): parse_create_definition() - (NULL)
                    ENTER: parse_destroy_definition(): 
                        ENTER: post_token_queue(): 
                        RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                        ENTER: reset_token_queue(): 
                        RETURN: reset_token_queue(): 
                    RETURN(node): parse_destroy_definition() - (NULL)
                    ENTER: parse_var_definition(): 
                        ENTER: post_token_queue(): 
                        RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                        ENTER: reset_token_queue(): 
                        RETURN: reset_token_queue(): 
                    RETURN(node): parse_var_definition() - (NULL)
                    ENTER: reset_token_queue(): 
                    RETURN: reset_token_queue(): 
                RETURN(node): parse_namespace_item() - (NULL)
                ENTER: parse_import_statement(): 
                RETURN(node): parse_import_statement() - (NULL)
                ENTER: parse_start_function(): 
                    ENTER: post_token_queue(): 
                    RETURN((void*)tqueue->crnt): post_token_queue() - (!NULL)
                    ENTER: advance_token(): 
                    RETURN(tqueue->crnt->tok): advance_token() - (!NULL)
                    ENTER: reset_token_queue(): 
                    RETURN: reset_token_queue(): 
                RETURN(node): parse_start_function() - (NULL)
                ENTER: reset_token_queue(): 
                RETURN: reset_token_queue(): 
            RETURN(node): parse_module_item() - (NULL)
(: "(": 7: 16: "badchar.k"
symbol: "this": 7: 20: "badchar.k"
.: ".": 7: 21: "badchar.k"
symbol: "is": 7: 23: "badchar.k"
.: ".": 7: 24: "badchar.k"
symbol: "a": 7: 25: "badchar.k"
.: ".": 7: 26: "badchar.k"
symbol: "name": 7: 30: "badchar.k"
+: "+": 7: 32: "badchar.k"
        RETURN(node): parse_module() - (NULL)
(: "(": 7: 16: "badchar.k"
symbol: "this": 7: 20: "badchar.k"
.: ".": 7: 21: "badchar.k"
symbol: "is": 7: 23: "badchar.k"
.: ".": 7: 24: "badchar.k"
symbol: "a": 7: 25: "badchar.k"
.: ".": 7: 26: "badchar.k"
symbol: "name": 7: 30: "badchar.k"
+: "+": 7: 32: "badchar.k"
    RETURN(node): parse() - (NULL)
    TRACE: main(): End Parse
    TRACE: main(): --------------------------------------------
    TRACE: main(): set trace state: true
    TRACE: main(): Begin Traverse
    ENTER: traverse_ast(): 
        TRACE: traverse_ast(): no ast present
    RETURN: traverse_ast(): 
    TRACE: main(): End Traverse
    TRACE: main(): --------------------------------------------
